var/spack/repos/builtin/packages/wps/package.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+import glob
+import tempfile
+from os.path import dirname,join
+import multiprocessing as mp
+
+class Wps(AutotoolsPackage):
+    """The Weather Research and Forecasting Pre-Processing System (WPS)
+    """
+
+    homepage = "https://www.mmm.ucar.edu/weather-research-and-forecasting-model"
+    #url      = "https://www2.mmm.ucar.edu/wrf/src/WPSV4.0.TAR.gz"
+    url      = "https://github.com/wrf-model/WPS/archive/v4.2.tar.gz"
+
+    version('4.2', sha256='3e175d033355d3e7638be75bc7c0bc0de6da299ebd175a9bbc1b7a121acd0168')
+
+    variant('build_type', default='dmpar',
+            values=('serial', 'serial_NO_GRIB2', 'dmpar', 'dmpar_NO_GRIB2'))
+
+    patch('patches/4.2/arch.Config.pl.patch', when='@4.2')
+    patch('patches/4.2/arch.configure.defaults.patch', when='@4.2')
+    patch('patches/4.2/configure.patch', when='@4.2')
+    patch('patches/4.2/preamble.patch', when='@4.2')
+
+
+    depends_on('mpi')
+    depends_on('wrf')
+    # According to:
+    # http://www2.mmm.ucar.edu/wrf/users/docs/user_guide_v4/v4.0/users_guide_chap2.html#_Required_Compilers_and_1
+    # Section: "Required/Optional Libraries to Download"
+    # parallel netcdf should not be used
+    depends_on('netcdf-c+parallel-netcdf')
+    depends_on('parallel-netcdf')
+    depends_on('netcdf-fortran')
+    # not sure if +fortran is required, but seems like a good idea
+    depends_on('hdf5+fortran+hl')
+    # build script use csh
+    depends_on('tcsh', type=('build'))
+    # time is not installed on all systems b/c bash provides it
+    # this fixes that for csh install scripts
+    depends_on('time', type=('build'))
+
+    depends_on('autoconf', type='build')
+    depends_on('automake', type='build')
+    depends_on('m4', type='build')
+    depends_on('libtool', type='build')
+    depends_on('jasper')
+
+    def setup_environment(self, spack_env, run_env):
+        spack_env.set('NETCDF', self.spec['netcdf-c'].prefix)
+        # This gets used via the applied patch files
+        spack_env.set('NETCDFF', self.spec['netcdf-fortran'].prefix)
+        spack_env.set('JASPERINC', self.spec['jasper'].prefix.include)
+        spack_env.set('JASPERLIB', self.spec['jasper'].prefix.lib)
+
+        # This is a hack - we need to use WRF staging ("spack install -- wrf")
+        spack_env.set('WRF_DIR', join(glob.glob(join(dirname(dirname(self.stage.source_path)), "spack-stage-wrf-%s-*" % str(self.version)))[0], "spack-src"))
```

Comment:
The stage normally gets deleted after each installation. Could we copy the WRF source code to somewhere in the WRF installation prefix so that it can be used by WPS?

Reply:
That's why I mentioned, as a comment on #9851, that issue #8054 is giving us grief. Not only does WPS need the WRF folder to build, users often need numerous files out of the WRF folder (and not just the resulting executables, e.g. `wrf.exe`) to run. For now, we're building WRF with `--keep-stage`, but as soon as we build WPS - Spack removes WRF's staging folder - which defeats the purpose (our ugly workaround: copy aside and later restore WRF's staging folder...). I might ask @MichaelLaufer to fix #8054, but I realize that's not a good long-term solution for building/using WRF and WPS.

Any advice?

Reply:
I would just copy the WRF source code to somewhere in the WRF installation prefix.

Reply:
@adamjstewart do you know of any other Spack packages with this type of dependence on the staging folder of a second package? An example would give us a good starting point to make the changes here.

Reply:
It looks like `tcl` and `tk` do this:
```python
            # Copy source to install tree                  
            installed_src = join_path(                                          
                self.spec.prefix, 'share', self.name, 'src')                    
            stage_src = os.path.realpath(self.stage.source_path)                
            install_tree(stage_src, installed_src)     
```
`occa` does something even simpler:
```python
        # Copy the source to the installation directory and build OCCA there.   
        install_tree('.', prefix)
```
Another option would be to add a `resouce(...)` directive to download the source code for the same version of WRF used to build WPS. Basically, if you can figure out which files you need to build WPS, I can tell you how to make sure those files stick around.